odl/discr/discr_mappings.py

```diff
@@ -35,11 +35,11 @@
 from odl.space.base_ntuples import NtuplesBase, FnBase
 from odl.space.fspace import FunctionSet, FunctionSpace
 from odl.set.domain import IntervalProd
+from odl.util.utility import is_valid_input_meshgrid
 
 
 __all__ = ('FunctionSetMapping',
-           'RawGridCollocation', 'GridCollocation', 'RawNearestInterpolation',
```

Comment:
Duck-typing. If the restriction/extension operator are defined on a `FunctionSpace`, they are linear. Otherwise, they're not. Apart from that, there is no fundamental difference. `GridCollocation` and colleagues now work in both cases.